modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -240,6 +247,18 @@ private CompletableFuture<LeaseGrantedMessageResponse> processLeaseGrantedMessag
         }));
     }
 
+    /**
+     * Process {@link WaitReplicaStateMessage}.
+     *
+     * @param msg Message to process.
+     * @return Future that contains a result.
+     */
+    private CompletableFuture<Void> processWaitReplicaStateMessage(WaitReplicaStateMessage msg) {
+        LOG.info("Received LeaseGrantedMessage for replica belonging to group=" + groupId());
+
+        return waitForActualState(FastTimestamps.coarseCurrentTimeMillis() + TimeUnit.SECONDS.toMillis(msg.timeout()));
```

Comment:
It's also required to sendPrimaryReplicaChangeToReplicationGroup after waitForActualState. Please check `Replica#processLeaseGrantedMessage` for more details.